icu4c/source/common/uarrsort.cpp

```diff
@@ -238,27 +232,19 @@ subQuickSort(char *array, int32_t start, int32_t limit, int32_t itemSize,
 static void
 quickSort(char *array, int32_t length, int32_t itemSize,
             UComparator *cmp, const void *context, UErrorCode *pErrorCode) {
-    alignas(max_align_t) char xw[2*STACK_ITEM_SIZE];
-    void *p;
-
     /* allocate two intermediate item variables (x and w) */
-    if(itemSize<=STACK_ITEM_SIZE) {
-        p=xw;
-    } else {
-        p=uprv_malloc(2*itemSize);
-        if(p==NULL) {
-            *pErrorCode=U_MEMORY_ALLOCATION_ERROR;
-            return;
-        }
+    icu::MaybeStackArray<max_align_t, sizeInMaxAlignTs(STACK_ITEM_SIZE)> x;
```

Comment:
I would prefer still making only one xw of twice an item size so that we heap-allocate at most once, and for w use p + sizeInMaxAlignTs(itemSize) where p = xw.getAlias().